WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -99,37 +99,43 @@ class RegisterDomainCoordinator {
     /// Related to the `purchaseFromDomainManagement` Domain selection type.
     /// Adds the selected domain to the cart then presents a site picker view.
     func handleExistingSiteChoice(on viewController: UIViewController) {
-        let config = BlogListConfiguration(shouldShowCancelButton: false,
-                                           shouldShowNavBarButtons: false,
-                                           navigationTitle: TextContent.sitePickerNavigationTitle,
-                                           backButtonTitle: TextContent.sitePickerNavigationTitle,
-                                           shouldHideSelfHostedSites: true,
-                                           shouldHideBlogsNotSupportingDomains: true)
+        let config = BlogListConfiguration(
+            shouldShowCancelButton: false,
+            shouldShowNavBarButtons: false,
+            navigationTitle: TextContent.sitePickerNavigationTitle,
+            backButtonTitle: TextContent.sitePickerNavigationTitle,
+            shouldHideSelfHostedSites: true,
+            shouldHideBlogsNotSupportingDomains: true,
+            analyticsSource: analyticsSource
+        )
         let blogListViewController = BlogListViewController(configuration: config, meScenePresenter: nil)
 
         blogListViewController.blogSelected = { [weak self] controller, selectedBlog in
             guard let self else {
                 return
             }
-            self.site = selectedBlog
             controller.showLoading()
             self.createCart { [weak self] result in
+                guard let self else {
+                    return
+                }
                 switch result {
                 case .success(let domain):
-                    self?.domainAddedToCartAndLinkedToSiteCallback?(controller, domain.domainName, selectedBlog)
-                    controller.hideLoading()
+                    self.site = selectedBlog
+                    self.domainAddedToCartAndLinkedToSiteCallback?(controller, domain.domainName, selectedBlog)
+                    self.track(.purchaseDomainSiteSelected)
```

Comment:
@hassaanelgarem I changed this a bit, the `site` is set and the event is fired when `createCart` is successful and the "Site Switcher" screen is dismissed.

You can test this scenario by:
- Navigating to `All Domains > Choose Existing Site > Site Switcher`.
- Disable your Wifi.
- Select a site.
- An activity indicator will appear, then the request will fail.
- The Site Switcher won't be dismissed.

In this case, I thought it's weird to set the site and fire the event, because the site wasn't really selected.

That being said, the `site_switcher_site_selected` is fired immediately when the row is selected while the event `domain_management_purchase_domain_site_selected` is fired only when the cart is created and is trully selected.

Reply:
I might be missing some context here so take it with a grain of salt. The distinction makes sense to me. However, will this information be useful when we look at the tracks? If there's no action the user can take between, it sounds more useful as a debug tool which maybe isn't a perfect fit for a track event.

Reply:
@alpavanoglu Resolved in 68f7814b8888ec56da9ed9781662cef8c68fc6b9

> However, will this information be useful when we look at the tracks?
Not so much, but if the request fails, this event won't be fired and it'll give us a hint that something is wrong. But as you said, it's more of a debug thing that analytics. I chatted with @hassaanelgarem and he supported the idea of removing that event. The `site_switched_site_selected` is enough.